app/assets/javascripts/map.js

```diff
@@ -1,152 +1,165 @@
 //= require polyline
 
-      var directionsDisplay;
-      var directionsService = new google.maps.DirectionsService();
-      var map;
+var directionsService = new google.maps.DirectionsService();
 
-      function initialize() {
-        directionsDisplay = new google.maps.DirectionsRenderer();
+function initialize() {
+  directionsDisplay = new google.maps.DirectionsRenderer();
 
-        var mapOptions = {
-          zoom: 4,
-          mapTypeId: google.maps.MapTypeId.ROADMAP
-        };
+  var mapOptions = {
+    zoom: 4,
+    mapTypeId: google.maps.MapTypeId.ROADMAP
+  };
 
-        map = new google.maps.Map(document.getElementById('map_canvas'), mapOptions);
-        directionsDisplay.setMap(map);
+  map = new google.maps.Map(document.getElementById('map_canvas'), mapOptions);
+  directionsDisplay.setMap(map);
+  route = new Route();
```

Comment:
Seems like you should `new` a Route and pass in two `Point` objects, from an OO perspective.